### PR TITLE
Scripts for building and testing the ATDM configure of trilinos on toss3 machines and on sems rhel6 machines

### PR DESCRIPTION
**Pull request by fryeguy52**
@trilinos/framework 

@bartlettroscoe can you review this?

## Description

Added scripts to build and test the ATDM configuration of trilinos on new machines.  This makes it easy to reproduce builds on these platforms.

## Related Issues

* Closes 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of 

## How Has This Been Tested?

The scripts were run on toss3 and rhel6 machines and were successful in building, testing, and submitting results to cdash.

## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers.
- [x] My code follows the code style of the affected package(s).
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] No new compiler warnings were introduced.
- [] These changes break backwards compatibility.

### CODE REVIEW

**Comment by mhoemmen** [comment]
Edited to tag the correct team (@trilinos/framework)

File: cmake/std/atdm/README.md [hunk 1]
```diff
@@ -254,6 +256,57 @@ $ srun ./checkin-test-sems.sh intel-opt-openmp \
...
+$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh intel-opt-openmp
+
+$ cmake \
+  -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
+  -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_MueLu=ON \
+  $TRILINOS_DIR
+
+$ make -j16
+
+$ cat > job.sh << "EOF"
```
Comment (bartlettroscoe) [inline cmake/std/atdm/README.md]: @fryeguy52,

Is it necessary to create this temp `job.sh` script?    Is it possible to simply just run:

```
srun -N 1 --time=600 --account=<YOUR_WCID> ctest -j16
```

?

File: cmake/std/atdm/utils/get_known_system_name.sh [hunk 2]
```diff
@@ -29,9 +29,21 @@ elif [[ $THIS_HOSTNAME == "white"* ]] ; then
 elif [[ $THIS_HOSTNAME == "ride"* ]] ; then
   ATDM_HOSTNAME=ride
   ATDM_SYSTEM_NAME=ride
+elif [[ $THIS_HOSTNAME == "chama"* ]] ; then
+  ATDM_HOSTNAME=chama
+  ATDM_SYSTEM_NAME=toss3
+elif [[ $THIS_HOSTNAME == "serrano"* ]] ; then
+  ATDM_HOSTNAME=serrano
+  ATDM_SYSTEM_NAME=toss3
+elif [[ -f /projects/sems/modulefiles/utils/get-platform ]] ; then
```
Comment (bartlettroscoe) [inline cmake/std/atdm/utils/get_known_system_name.sh]: Nice logic!

File: cmake/std/atdm/README.md [hunk 3]
```diff
@@ -254,6 +256,57 @@ $ srun ./checkin-test-sems.sh intel-opt-openmp \
...
+  #! /bin/bash \
+  ctest -j16 \
+  EOF
+$ chmod +x job.sh
+
+$ srun -N 1 --time=600 --account=<YOUR_WCID> -J $JOB_NAME  -e output.error ./job.sh &
```
Comment (bartlettroscoe) [inline cmake/std/atdm/README.md]: @fryeguy52, can we say something about the argument `-account=<YOUR_WCID>` and explain where `<YOUR_WCID>` comes from?  We can not give away any sensitive SNL info but we should say something about what this is and who to contact or where to go to find out what account to use and how to be added to that account.

**Comment by bartlettroscoe** [comment]
I tested this on 'serrano' as well with:

```
$ ./checkin-test-atdm.sh all --enable-all-packages=off --no-enable-fwd-packages --enable-packages=Kokkos --configure --build
```

and it returned:

```
PASSED (NOT READY TO PUSH): Trilinos: serrano-login2

Tue Apr 10 20:54:51 MDT 2018

Enabled Packages: Kokkos
Enabled all Forward Packages

Build test results:
-------------------
0) MPI_RELEASE_DEBUG_SHARED_PT => Test case MPI_RELEASE_DEBUG_SHARED_PT was not run! => Does not affect push readiness! (-1.00 min)
1) intel-debug-openmp => passed: build-only passed => Not ready to push! (4.36 min)
2) intel-opt-openmp => passed: build-only passed => Not ready to push! (4.66 min)
```

Once we address some of the documentation issues, I think this is ready to merge to 'develop' and then set up Jenkins jobs.

File: cmake/ctest/drivers/atdm/toss3/drivers/Trilinos-atdm-toss3-intel-debug-openmp-panzer.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+export SRUN_CTEST_TIME_LIMIT_MINUTES=300 # 5 hour time limit
+$WORKSPACE/Trilinos/cmake/ctest/drivers/atdm/toss3/local-driver.sh
```
Comment (bartlettroscoe) [inline cmake/ctest/drivers/atdm/toss3/drivers/Trilinos-atdm-toss3-intel-debug-openmp-panzer.sh]: @fryeguy52, for now, I think we need to disable the running of the tests on 'toss3' ('serrano' and 'chama') for with:

```
export CTEST_DO_TEST=OFF
```

right?

I will get the ctest driver working to run the tests on the compute node separately later.

**Comment by fryeguy52** [comment]
@bartlettroscoe I confirmed that a direct call to ctest through sbatch works and simplified the documentation.  I also add a note for SNL users on where to find information on WCIDs

File: cmake/std/atdm/README.md [hunk 5]
```diff
@@ -254,6 +256,55 @@ $ srun ./checkin-test-sems.sh intel-opt-openmp \
...
+$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh intel-opt-openmp
+
+$ cmake \
+  -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
+  -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_MueLu=ON \
+  $TRILINOS_DIR
+
+$ make -j16
+
+$ srun -N 1 --time=600 --account=<YOUR_WCID> -J $JOB_NAME  -e output.error ctest -j16
```
Comment (bartlettroscoe) [inline cmake/std/atdm/README.md]: @fryeguy52,

Can we take out `-e output.error`?   I found that just hides the error output and no one ever notices.

Comment (bartlettroscoe) [reply cmake/std/atdm/README.md]: Note that the file `ctest/cmake/drivers/atdm/toss3/local-driver.sh` does not have `-e output.error` so we should not need it here either.

File: cmake/ctest/drivers/atdm/toss3/drivers/Trilinos-atdm-toss3-intel-debug-openmp-panzer.sh (hunk 4 above)
Comment (fryeguy52) [reply cmake/ctest/drivers/atdm/toss3/drivers/Trilinos-atdm-toss3-intel-debug-openmp-panzer.sh]: export CTEST_DO_TEST=OFF 
is in the $WORKSPACE/Trilinos/cmake/ctest/drivers/atdm/toss3/local-driver.sh script.  Is it preferable to move it the individual scripts?

File: cmake/ctest/drivers/atdm/toss3/local-driver.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash -l
+
+if [ "${SRUN_CTEST_TIME_LIMIT_MINUTES}" == "" ] ; then
+  SRUN_CTEST_TIME_LIMIT_MINUTES=180  # Default limit is 3 hours
+fi
+
+set -x
+
+export CTEST_DO_TEST=OFF
```
Comment (bartlettroscoe) [inline cmake/ctest/drivers/atdm/toss3/local-driver.sh]: Okay, there it is.   I missed that the first time through.  Thanks!

File: cmake/ctest/drivers/atdm/toss3/drivers/Trilinos-atdm-toss3-intel-debug-openmp-panzer.sh (hunk 4 above)
Comment (bartlettroscoe) [reply cmake/ctest/drivers/atdm/toss3/drivers/Trilinos-atdm-toss3-intel-debug-openmp-panzer.sh]: Okay, I see it set there now.  That is a much better place for for this.  Thanks!

**Comment by fryeguy52** [comment]
@bartlettroscoe I just pushed a commit that removes "-e output.error" from the srun call

**Pull request merged**